groth16-framework/test_data/Groth16VerifierExtensions.sol

```diff
@@ -137,53 +155,75 @@ contract Query2 is Verifier {
         );
 
-        uint32 max_block_number = convertToU32(pis, PI_MAX_BLOCK_NUM_OFFSET);
+        uint32 maxBlockNumber = convertToU32(pis, PI_MAX_BLOCK_NUM_OFFSET);
         require(
-            max_block_number == query.maxBlockNumber,
+            maxBlockNumber == query.maxBlockNumber,
             "The parsed max block number must be equal to the expected one in query."
         );
 
-        address contract_address = convertToAddress(pis, PI_CONTRACT_ADDR_OFFSET);
+        address contractAddress = convertToAddress(pis, PI_CONTRACT_ADDR_OFFSET);
         require(
-            contract_address == query.contractAddress,
+            contractAddress == query.contractAddress,
             "The parsed contract address must be equal to the expected one in query."
         );
 
-        address user_address = convertToAddress(pis, PI_USER_ADDR_OFFSET);
+        address userAddress = convertToAddress(pis, PI_USER_ADDR_OFFSET);
         require(
-            user_address == query.userAddress,
+            userAddress == query.userAddress,
             "The parsed user address must be equal to the expected one in query."
         );
 
-        bytes32 block_hash = convertToHash(pis, PI_BLOCK_HASH_OFFSET);
+        bytes32 blockHash = bytes32(convertToHash(pis, PI_BLOCK_HASH_OFFSET));
         require(
-            block_hash == query.blockHash,
+            blockHash == query.blockHash,
             "The parsed block hash must be equal to the expected one in query."
         );
+
+        if (query.identifier == QUERY_IDENTIFIER_ERC20) {
+            uint256 rewardsRate = convertByteSliceToU256(pis, PI_REWARDS_RATE_OFFSET);
+            require(
+                rewardsRate == query.rewardsRate,
+                "The parsed rewards rate must be equal to the expected one in query."
+            );
+        }
+
+        require(
+            uint256(uint8(pis[PI_QUERY_IDENTIFIER_OFFSET])) == query.identifier,
+            "The parsed identifier must be equal to the expected one in query."
+        );
+    }
+
+    // Parse the query result from the plonky2 public inputs.
+    function parseQueryResult(bytes memory pis, uint256 identifier) internal pure returns (uint256[] memory) {
+        if (identifier == QUERY_IDENTIFIER_NFT) {
+            return parseNftIds(pis);
+        } else if (identifier == QUERY_IDENTIFIER_ERC20) {
+            return parseErc20Result(pis);
+        } else {
+            revert("Unsupported query identifier");
+        }
     }
 
     // Parse the `L` NFT IDs from the plonky2 public inputs.
     function parseNftIds(bytes memory pis) internal pure returns (uint256[] memory) {
         uint256[] memory nft_ids = new uint256[](L);
         for (uint32 i = 0; i < L; ++i) {
-            nft_ids[i] = uint256(convertToLeftPaddingU32(pis, PI_NFT_IDS_OFFSET + i * 8));
+            nft_ids[i] = uint256(convertToU32(pis, PI_NFT_IDS_OFFSET + i * 8));
```

Comment:
The format of the results in the NFT proofs are changing as well ?

Reply:
I think the previous one is not right, because I update to [decode the NFT IDs by the ABI](https://github.com/Lagrange-Labs/Euclid-database/blob/4a1defd2fcaefe464b1992278208d1fd3e81fbed/groth16-framework/tests/queries.rs#L120) in test code, and [assert the expected result](https://github.com/Lagrange-Labs/Euclid-database/blob/4a1defd2fcaefe464b1992278208d1fd3e81fbed/groth16-framework/tests/queries.rs#L134). It should be this sequence.

Could you also help check in lagrange-lpn-contracts as [this old comment](https://github.com/Lagrange-Labs/Euclid-database/blob/ffd966a05aaf32d270bdc5a9a5efac077f174c4c/groth16-framework/test_data/Groth16VerifierExtensions.sol#L220)? Thanks.

Reply:
After discussion, it should not be changed, let me revert to the previous version.

Reply:
Try to fix it in PR https://github.com/Lagrange-Labs/Euclid-database/pull/24.

Reply:
I merged the above PR, and the commit is [e0a1e13](https://github.com/Lagrange-Labs/Euclid-database/pull/21/commits/e0a1e13ee68fa05429e5fda373807daaa0570872).